app/controllers/navigation/CacheDependentNavigators.scala

```diff
@@ -211,60 +211,60 @@ trait CacheDependentNavigators {
+  protected def departureTransportPreviousPage(cacheModel: ExportsDeclaration): Call = {
+    val gotoInlandTransport = inlandOrBorderHelper.skipInlandOrBorder(cacheModel) || !cacheModel.isInlandOrBorder(Border)
+    if (gotoInlandTransport) routes.InlandTransportDetailsController.displayPage else routes.InlandOrBorderController.displayPage
   }
 
+  protected def departureTransportPreviousPageOnClearance(cacheModel: ExportsDeclaration): Call =
+    // For Clearance the previous page should always be SupervisingCustomsOffice by any means, as
+    // "1040" as PC is not applicable to this journey type.
+    routes.SupervisingCustomsOfficeController.displayPage
+
   protected def transportCountryPreviousPage(cacheModel: ExportsDeclaration): Call =
     if (cacheModel.isInlandOrBorder(InlandOrBorder.Border)) routes.DepartureTransportController.displayPage
     else routes.BorderTransportController.displayPage
 
-  protected def expressConsignmentPreviousPageOnStandard(cacheModel: ExportsDeclaration): Call = {
-    val backToInlandTransport = isPostalOrFTIModeOfTransport(cacheModel.inlandModeOfTransportCode) || isGuernseyOrJerseyDestination(cacheModel)
-    if (backToInlandTransport) routes.InlandTransportDetailsController.displayPage
-    else {
+  protected def expressConsignmentPreviousPage(cacheModel: ExportsDeclaration): Call = {
+    val guernseyOrJersey = isGuernseyOrJerseyDestination(cacheModel)
+    if (cacheModel.isInlandOrBorder(Border)) {
       val postalOrFTI = isPostalOrFTIModeOfTransport(cacheModel.transportLeavingBorderCode)
-      if (postalOrFTI && cacheModel.isInlandOrBorder(Border)) routes.InlandOrBorderController.displayPage
+      if (postalOrFTI || guernseyOrJersey) routes.InlandOrBorderController.displayPage
+      else routes.TransportCountryController.displayPage
+    } else {
+      val postalOrFTI = isPostalOrFTIModeOfTransport(cacheModel.inlandModeOfTransportCode)
+      if (postalOrFTI || guernseyOrJersey) routes.InlandTransportDetailsController.displayPage
       else routes.TransportCountryController.displayPage
     }
   }
 
-  protected def expressConsignmentPreviousPageOnClearance(cacheModel: ExportsDeclaration): Call = {
-    val postalOrFTI = isPostalOrFTIModeOfTransport(cacheModel.transportLeavingBorderCode)
-    if (postalOrFTI && supervisingCustomsOfficeHelper.checkProcedureCodes(cacheModel)) routes.WarehouseIdentificationController.displayPage
-    else if (postalOrFTI) routes.SupervisingCustomsOfficeController.displayPage
-    else routes.DepartureTransportController.displayPage
-  }
+  protected def expressConsignmentPreviousPageOnClearance(cacheModel: ExportsDeclaration): Call =
+    if (!isPostalOrFTIModeOfTransport(cacheModel.transportLeavingBorderCode)) routes.DepartureTransportController.displayPage
+    // For Clearance the previous page should always be SupervisingCustomsOffice by any means, as
+    // "1040" as PC is not applicable to this journey type. This test is sort of superfluous then.
```

Comment:
Does this comment belong here? :thinking:

Reply:
This test was already here and, as per comment, is actually superfluous. But since it does not harm, it's just an additional test, and instead to remove it I left a comment.